rust/src/jni_utils.rs

```diff
@@ -440,6 +468,48 @@ pub(crate) unsafe fn f64_from_jobject(obj: jobject, jni_env: *mut JNIEnv) -> err
     }
 }
 
+macro_rules! primitive_array_from_jobject {
+    ($fn_name:ident, $rust_type:ty, $get_array_element:path, $release_array_element:path) => {
+        pub(crate) unsafe fn $fn_name(obj: jobject, jni_env: *mut JNIEnv) -> errors::Result<Vec<$rust_type>> {
+            if obj.is_null() {
+                Err(errors::J4RsError::JniError(
+                    format!("Attempt to create an {} array from null", stringify!($rust_type)),
+                ))
+            } else {
+                let length = (opt_to_res(cache::get_jni_get_array_length())?)(
+                    jni_env,
+                    obj
+                );
+                let bytes = (opt_to_res($get_array_element())?)(
```

Comment:
Could we change the identifier to something that includes every type the PR deals with? Maybe something like `vals` or something...

Reply:
renamed to `val_ptr` to signify that it's a raw pointer